### ISSUE TITLE
Direct Message Endpoints Deprecation: June 19th

### ISSUE THREAD

**Issue by daviddannunzio** [ryanmcgrath/twython#489]
Hello,

I currently use the Twython library and saw in [this](https://twittercommunity.com/t/upcoming-changes-to-the-developer-platform/104603) recent post that the following direct message endpoints will no longer be functional after June 19th. 

* GET direct_messages
* GET direct_messages/sent
* GET direct_messages/show
* POST direct_messages/destroy

Are there plans to update Twython to migrate to the new functional endpoints before June 19th?

**Comment by michaelhelmick** [comment]
Currently, no. But if you’d be able to open a pr with the changes, I’d be more than happy to merge it!

**Comment by decklord** [comment]
This is a critical issue

**Comment by andremann** [comment]
+1
The same will happen for the userstream and sitestream APIs
Major issues on the way!!

**Comment by andremann** [comment]
Twitter will phase out the API later!! https://twitter.com/TwitterDev/status/982346370882461696

**Comment by michaelhelmick** [comment]
For reference, we'll need to use endpoints mentioned here https://developer.twitter.com/en/docs/accounts-and-users/subscribe-account-activity/api-reference/aaa-premium

**Comment by andremann** [comment]
For the record, I was using user stream API in order to get @mentions and I refactored my code to use the mentions_timeline API [1] which is accessible with standard privileges. Since it is rate limited, I poll the endpoint every 15 seconds and this is enough to keep it going without triggering the limiter.
I imagine something similar can be done for direct messages [2].

Avoiding the stream API is rather inconvenient and the solution is not elegant, but it works!
Basically, I save in a file the id of the most recent mentions and then every 15 seconds I ask for new ones availing of the `since_id` param.
The code for my Twitter bot can be seen here [3].

[1] https://developer.twitter.com/en/docs/tweets/timelines/api-reference/get-statuses-mentions_timeline
[2] https://developer.twitter.com/en/docs/direct-messages/api-features
[3] https://github.com/andremann/neni-the-ambulance

**Comment by cepradeep** [comment]
I'd be happy to contribute the changes :)

I briefly looked through the repo, and was unable to find any docs / steps to follow for contributing. So if there is anything specific I've to keep in mind, please point me to it. Thanks!
